lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -459,28 +457,30 @@ def access_token_url
       end
 
       def setup_access_token
-        request = 'grant_type=client_credentials'
-        begin
-          raw_response = ssl_post(access_token_url, request, access_token_header)
-        rescue ResponseError => e
-          raise OAuthResponseError.new(e)
-        else
-          response = parse(raw_response)
+        response = parse(ssl_post(access_token_url, 'grant_type=client_credentials', access_token_header))
+        @options[:access_token] = response['access_token']
+        @options[:expires] = (DateTime.now + response.fetch('expires_in', 0).seconds).strftime('%Q').to_i
 
-          if (access_token = response['access_token'])
-            access_token
-          else
-            raise OAuthResponseError.new(response)
-          end
-        end
+        Response.new(
+          response['access_token'].present?,
+          message_from(true, response, {}),
+          response,
+          test: test?,
+          error_code: error_code_from(false, response, {})
+        )
+      rescue ResponseError => e
+        Response.new(false, e.response.body.to_s, {}, test: test?)
       end
 
-      def commit(action, post, options, authorization = nil, method = :post)
+      def access_token_valid?
+        @options[:access_token] && @options.fetch(:expires, 0) > DateTime.now.strftime('%Q').to_i
```

Comment:
I would suggest making sure the the default value for `expires` is 0. Currently `@options.fetch(:expires, 0)` could be nil which cause a NoMethodError when checking if it's less than

Reply:
Oh it think that's what you are trying to do with fetch and make 0 the default but for some reason it's not working

Reply:
I will change it to:

`@options[:access_token] && @options[:expires] && @options[:expires] > DateTime.now.strftime('%Q').to_i`

Reply:
done!